delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/ResponseToJsonType.scala

```diff
@@ -0,0 +1,40 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.directives
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.MediaType
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.DeltaDirectives.unacceptedMediaTypeRejection
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.Response.{Complete, Reject}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.HeadersUtils
+import io.circe.Json
+import monix.bio.UIO
+import monix.execution.Scheduler
+
+sealed trait ResponseToJsonType {
+  def apply(): Route
+}
+
+object ResponseToJsonType {
```

Comment:
This way is fine, but then you're going to have to add the instances here:
```scala
object ResponseToJsonType extends JsonValueInstances with SparqlValueInstances
```

when you have that in place, you can and should remove the explicit import

Reply:
I can't because `SparqlValueInstances` instances are in blazegraph plugin and this is in `sdk`

Reply:
Ah, then you yo have do do it differently. Simplify ResponseToJsonType to be just an object that returns the route:

```scala
object ResponseToJsonType {
	apply(): Route = ???
}
```

And call it from ResponseToJson and ResponseToSparql.

And then have something like `BlazegraphDirectives` where you have your own emit:
```
object BlazegraphDirectives {

  /**
    * Completes the current Route with the provided conversion to SparqlResults
    */
  def emit(response: ResponseToSparql): Route =
    response()
}
```

Reply:
Not sure I understand what's the `None` here.

Reply:
Sorry, you don't need None